home/wagtail_hooks.py

```diff
@@ -38,3 +42,16 @@ def sort_footer_page_listing_by_path(parent_page, pages, request):
         pages = pages.order_by('path')
 
     return pages
+
+
+@hooks.register('construct_page_chooser_queryset')
+def limit_page_chooser(pages, request):
```

Comment:
It is not clear enough to me what is happening in this function. There needs to be tests for this at the very least. Perhaps an explanatory comment. Ideally, we should only intervene in the case where we're selecting a page for a Section's featured content, otherwise, the queryset should not be touched.

Reply:
I have added comments and tests and simplified some code kindly check @istride